Add OctoAI LLM endpoint wrapper and embedding endpoint wrapper

Bassem-Yacoube opened the pull request:
This pull request introduces support for [OctoAI's Compute Service](https://octoai.cloud/) embedding endpoints and large language model endpoints into the LangChain project. The changes encompass additions to various files to accommodate OctoAI's API integration.

**Details of the changes:**

**langchain/__init__.py** and **langchain/llms/__init__.py:** Updates to these initialization files allow OctoAI's services to be utilized in the LangChain's workflow. The changes add 'OctoAIEndpoint' as an additional language model option.

**langchain/embeddings/octoai_embeddings.py:** This new file creates a wrapper for OctoAI embedding endpoints. This addition allows users to compute embeddings of documents and queries using OctoAI's service.

**langchain/llms/octoai_endpoint.py**: This file introduces a new class, 'OctoAIEndpoint', which serves as a wrapper for OctoAI's large language model endpoints. It includes methods for interacting with OctoAI's API and getting responses from it.

**tests/integration_tests/embeddings/test_octoai_embeddings.py**: This file introduces tests to ensure the correct functioning of the OctoAI embedding endpoint using the Instructor-Large model endpoint. It includes tests for document and query embedding functionalities.

**tests/integration_tests/llms/test_octoai_endpoint.py**: This file includes tests for OctoAI's LLM wrapper for MPT-7B model endpoint. It covers various scenarios including valid API calls, calls that result in errors, and the saving/loading of language model endpoints.

**docs/modules/models/llms/integrations/octoai.ipynb**: This notebook shows how to use LangChain to interact with `OctoAI` [LLM endpoints](https://octoai.cloud/templates)

Please note that to use the newly added OctoAI's services, users should have the octoai python package installed `pip install octoai`, and the environment variables OCTOAI_API_TOKEN set with the API token and ENDPOINT_URL set with endpoint_url. The API token and endpoint URL can be obtained by cloning a template and creating an API token at https://octoai.cloud.

Thanks in advance for your review!

@hwchase17 @agola11

Commit:
add octoai_endpoint llm

Commit:
Merge branch 'master' of https://github.com/octoml/langchain

Commit:
fix example notebook

Bassem-Yacoube committed:
Merge pull request #1 from octoml/octoai

OctoAI Endpoint

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
add embeddings using instructor large endpoint

Bassem-Yacoube committed:
Merge pull request #2 from octoml/octoai

add embeddings using instructor large endpoint

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

hwchase17 commented on langchain/__init__.py:
```diff
@@ -29,6 +29,7 @@
     HuggingFaceTextGenInference,
     LlamaCpp,
     Modal,
+    OctoAIEndpoint,
```
we probably shouldnt add it here

Bassem-Yacoube committed:
Update __init__.py

Bassem-Yacoube committed:
Update __init__.py

Bassem-Yacoube reviewed (commented):
Yes, that works. 
Thank you.

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
fix lint errors

AI-Bassem commented:
I fixed the lint errors but Vercel seems to still be failing. I can't see what the error is with Vercel. Can you please let me know what needs to be fixed?

Bassem-Yacoube committed:
Merge branch 'master' into master

Commit:
fix embeddings signatures and tests

AI-Bassem commented:
conflicts resolved. lint and test errors fixed. Vercel deployment now succeeding.

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
add octoai-sdk dependency to pyproject.toml

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
remove octoai-sdk

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
add octoai-sdk dependency

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
update lock file by running poetry lock --no-update

Commit:
resolve conflict for content-hash

Bassem-Yacoube committed:
Merge branch 'master' into master

Bassem-Yacoube committed:
Merge branch 'master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Commit:
fix formatting and toml

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Bassem-Yacoube committed:
Merge branch 'master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master

shotarok committed:
Update `langchain.chains.create_extraction_chain_pydantic` to parse results successfully (#6887)

<!-- Thank you for contributing to LangChain!

Replace this comment with:
  - Description: a description of the change, 
  - Issue: the issue # it fixes (if applicable),
  - Dependencies: any dependencies required for this change,
- Tag maintainer: for a quicker response, tag the relevant maintainer
(see below),
- Twitter handle: we announce bigger features on Twitter. If your PR
gets announced and you'd like a mention, we'll gladly shout you out!

If you're adding a new integration, please include:
1. a test for the integration, preferably unit tests that do not rely on
network access,
  2. an example notebook showing its use.

Maintainer responsibilities:
  - General / Misc / if you don't know who to tag: @dev2049
  - DataLoaders / VectorStores / Retrievers: @rlancemartin, @eyurtsev
  - Models / Prompts: @hwchase17, @dev2049
  - Memory: @hwchase17
  - Agents / Tools / Toolkits: @vowelparrot
  - Tracing / Callbacks: @agola11
  - Async: @agola11

If no one reviews your PR within a few days, feel free to @-mention the
same people again.

See contribution guidelines for more information on how to write/run
tests, lint, etc:
https://github.com/hwchase17/langchain/blob/master/.github/CONTRIBUTING.md
 -->
 
- Description: 
- The current code uses `PydanticSchema.schema()` and
`_get_extraction_function` at the same time. As a result, a response
from OpenAI has two nested `info`, and
`PydanticAttrOutputFunctionsParser` fails to parse it. This PR will use
the pydantic class given as an arg instead.
- Issue: no related issue yet
- Dependencies: no dependency change
- Tag maintainer: @dev2049
- Twitter handle: @shotarok28

rian-dolphin committed:
add with score option for max marginal relevance (#6867)

### Adding the functionality to return the scores with retrieved
documents when using the max marginal relevance
- Description: Add the method
`max_marginal_relevance_search_with_score_by_vector` to the FAISS
wrapper. Functionality operates the same as
`similarity_search_with_score_by_vector` except for using the max
marginal relevance retrieval framework like is used in the
`max_marginal_relevance_search_by_vector` method.
  - Dependencies: None
  - Tag maintainer: @rlancemartin @eyurtsev 
  - Twitter handle: @RianDolphin

---------

Co-authored-by: Dev 2049 <dev.dev2049@gmail.com>

shashankdeshpande committed:
added example notebook - use custom functions with openai agent (#6865)

<!-- Thank you for contributing to LangChain!

Replace this comment with:
  - Description: a description of the change, 
  - Issue: the issue # it fixes (if applicable),
  - Dependencies: any dependencies required for this change,
- Tag maintainer: for a quicker response, tag the relevant maintainer
(see below),
- Twitter handle: we announce bigger features on Twitter. If your PR
gets announced and you'd like a mention, we'll gladly shout you out!

If you're adding a new integration, please include:
1. a test for the integration, preferably unit tests that do not rely on
network access,
  2. an example notebook showing its use.

Maintainer responsibilities:
  - General / Misc / if you don't know who to tag: @dev2049
  - DataLoaders / VectorStores / Retrievers: @rlancemartin, @eyurtsev
  - Models / Prompts: @hwchase17, @dev2049
  - Memory: @hwchase17
  - Agents / Tools / Toolkits: @vowelparrot
  - Tracing / Callbacks: @agola11
  - Async: @agola11

If no one reviews your PR within a few days, feel free to @-mention the
same people again.

See contribution guidelines for more information on how to write/run
tests, lint, etc:
https://github.com/hwchase17/langchain/blob/master/.github/CONTRIBUTING.md
 -->

Bassem-Yacoube committed:
Merge branch 'hwchase17:master' into master
